mllib/src/test/scala/org/apache/spark/ml/feature/ImputerSuite.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml.feature
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.ml.util.{DefaultReadWriteTest}
+import org.apache.spark.mllib.util.MLlibTestSparkContext
+import org.apache.spark.mllib.util.TestingUtils._
+import org.apache.spark.sql.Row
+
+class ImputerSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultReadWriteTest {
+
+  test("Imputer for Double with default missing Value NaN") {
+    val df = sqlContext.createDataFrame( Seq(
+      (0, 1.0, 1.0, 1.0, 1.0),
+      (1, 1.0, 1.0, 1.0, 1.0),
+      (2, 3.0, 3.0, 3.0, 3.0),
+      (3, 4.0, 4.0, 4.0, 4.0),
+      (4, Double.NaN, 2.25, 1.0, 1.0 )
+    )).toDF("id", "value", "mean", "median", "mode")
+    Seq("mean", "median", "mode").foreach { strategy =>
+      val imputer = new Imputer().setInputCol("value").setOutputCol("out").setStrategy(strategy)
+      val model = imputer.fit(df)
+      model.transform(df).select(strategy, "out").collect()
+        .foreach { case Row(d1: Double, d2: Double) =>
+          assert(d1 ~== d2 absTol 1e-5, s"Imputer ut error: $d2 should be $d1")
```

Comment:
Could we make this error message more informative, like "Imputed values differ. Expected: $d1, actual: $d2"